pkg/find-interface.go

```diff
@@ -0,0 +1,105 @@
+package impl
```

Comment:
Again, personal preference - I wouldn't use the `pkg` package or directory name. It doesn't convey any more meaning than `util` or `code`. In my linters I use `check`, and I put all the logic there (the checker). Up to you what the name should be.

Either way, the directory name and package name should match. `import impl "foo.bar/impl/pkg"` seems nasty to me.

Think of `go/ast` or `go/parser`, for example. Their package names don't contain the name `go` either.

Reply:
Agreed. To be honest, I have no idea why I didn't at least throw it into a `pkg/impl` directory. My primary goal was to separate out the CLI implementation package from the library, though maybe it would make more sense to move the CLI `main` package into`cmd/impl`. Thoughts on moving the lib to `impl` versus moving main to `cmd/impl`?

Reply:
I find that moving the implementation into an `impl` package is also a tad redundant. You're going to end up with verbose import paths like `impl/impl`, or `impl/pkg/impl`.

If you want the root package to be the main package, I'd call the library sub-package something more meaningful. I suggested `check` above, but you could go with `linter` or any other descriptive name you fancy.

I used to go the `foo/cmd/foo` route myself (e.g. `github.com/mvdan/interfacer/cmd/interfacer`), but I've grown to dislike that mechanism. Especially for repositories that host a single tool. The import path that 99% of people will use becomes terribly long and verbose. Now they download `mvdan.cc/interfacer`, and those 1% that want the library can just use `mvdan.cc/interfacer/check`.